libvast/src/table_slice.cpp

```diff
@@ -363,18 +364,14 @@ std::shared_ptr<arrow::RecordBatch> as_record_batch(const table_slice& slice) {
         // lifetime with the chunk.
         auto batch = state(encoded, slice.state_)->record_batch();
         auto result = std::shared_ptr<arrow::RecordBatch>{
-          batch.get(), [slice, batch](arrow::RecordBatch*) noexcept {
-            VAST_IGNORE_UNUSED(slice);
-            VAST_IGNORE_UNUSED(batch);
-          }};
+          batch.get(), detail::keeper{batch, slice}};
         return result;
       } else {
         // Rebuild the slice as an Arrow-encoded table slice.
         auto copy = rebuild(slice, table_slice::encoding::arrow);
         // Bind the lifetime of the copy (and thus the returned Record Batch) to
         // the lifetime of the original slice.
-        slice.chunk_->add_deletion_step(
-          [copy]() noexcept { VAST_IGNORE_UNUSED(copy); });
+        slice.chunk_->add_deletion_step(detail::keeper{copy});
```

Comment:
The comment gives a great suggestion for a better name: `bind` or `bind_lifetime`. That would be a free function that returns a `binder`.

Reply:
Now that I'm reading it again `detail::keep(copy)` isn't so bad either.

Reply:
I actually like `keeper` a lot. Not sure if a function like this really adds any value:

```cpp
template <class... Args>
constexpr auto keep(Args&&... args) noexcept {
  return keeper{std::forward<Args>(args)...};
}
```

Reply:
Will implement my above comment as discussed in Slack.